include/libtorrent/alert_types.hpp

```diff
@@ -66,6 +66,15 @@ namespace libtorrent
 	// user defined alerts should use IDs greater than this
 	static const int user_alert_id = 10000;
 
+	static const int base_alert_id_counter = __COUNTER__ + 1;
```

Comment:
with a constexpr counter, is the ordering and assignment of ids guaranteed to be deterministic? I imagine it would depend on the order of instantiating templates. I think it's important, especially for ABI compatibility, that the assignment of IDs is deterministic and identical to the ``COUNTER`` preprocessor.